R/test-compiled-code.r

```diff
@@ -0,0 +1,131 @@
+#' Test Compiled Code in a Package
+#'
+#' Test compiled code in the package \code{package}. See
+#' \code{\link{use_catch}()} for more details.
+#'
+#' @param package The name of the package to test.
+#'
+#' @export
+test_compiled_code <- function(package) {
+
+  routine <- get_routine(package, "run_testthat_tests")
+
+  output <- ""
+  tests_passed <- TRUE
+
+  tryCatch(
+    output <- utils::capture.output(tests_passed <- .Call(routine)),
+    error = function(e) {
+      warning(sprintf("failed to call test entrypoint '%s'", routine))
+    }
+  )
+
+  # Drop first line of output (it's jut a '####' delimiter)
+  info <- paste(output[-1], collapse = "\n")
+
+  expect_that(tests_passed, function(result) {
```

Comment:
Can you please look at `expect_silent()` to see how a modern expectation should look?